xugrid/ugrid/ugrid2d.py

```diff
@@ -1319,14 +1339,37 @@ def sel_points(self, obj, x: FloatArray, y: FloatArray):
         dim = self.face_dimension
         xy = np.column_stack([x, y])
         index = self.locate_points(xy)
+
+        keep = slice(None, None)  # keep all by default
+        condition = None
         valid = index != -1
-        index = index[valid]
+        if not valid.all():
+            msg = "Not all points are located inside of the grid."
+            if out_of_bounds == "raise":
+                raise ValueError(msg)
+            elif out_of_bounds in ("warn", "ignore"):
+                if out_of_bounds == "warn":
+                    warnings.warn(msg)
+                condition = xr.DataArray(valid, dims=(dim,))
+            elif out_of_bounds == "drop":
+                index = index[valid]
+                keep = valid
+            else:
+                # This code shouldn't be reachable due to check up top.
```

Comment:
In that case remove it? You also implemented a testcase that checks if a faulty string raises that error that you refer to here, so even if someone will accidentally remove that code, they will see the testcase popping up.